include/arch/riscv32/common/linker.ld

```diff
@@ -36,9 +36,9 @@
 MEMORY
 {
 #ifdef CONFIG_XIP
-    ROM (rx)  : ORIGIN = CONFIG_RISCV_ROM_BASE_ADDR, LENGTH = CONFIG_RISCV_ROM_SIZE
+    ROM (rx)  : ORIGIN = RISCV_ROM_BASE, LENGTH = RISCV_ROM_SIZE
 #endif
-    RAM (rwx) : ORIGIN = CONFIG_RISCV_RAM_BASE_ADDR, LENGTH = RISCV_RAM_SIZE
+    RAM (rwx) : ORIGIN = RISCV_RAM_BASE, LENGTH = RISCV_RAM_SIZE
```

Comment:
Why do we need to come up with new, architecture-specific indirection variables for RAM and ROM base and size? Why not just rely on `DT_SRAM_BASE_ADDRESS` + `DT_RAM_SIZE` and friends directly?

Reply:
`RISCV_RAM_BASE`/`RISCV_RAM_SIZE` are not new things, they are currently defined in [openisa_rv32m1](https://github.com/zephyrproject-rtos/zephyr/blob/master/soc/riscv32/openisa_rv32m1/soc.h#L108), [sifive-freedom](https://github.com/zephyrproject-rtos/zephyr/blob/master/soc/riscv32/riscv-privilege/sifive-freedom/soc.h#L43), [miv](https://github.com/zephyrproject-rtos/zephyr/blob/master/soc/riscv32/riscv-privilege/miv/soc.h#L57) and referenced in: [libc-hooks.c](https://github.com/zephyrproject-rtos/zephyr/blob/master/lib/libc/newlib/libc-hooks.c#L46).

We just used them for `RAM` and, here you are totally right, added the analugous set of defines for `ROM`.

I agree that using `DT_` directly reduces the indirection level.
I can rework `linker.ld` and drop `RISCV_ROM_BASE`/`RISCV_RAM_BASE`, but I would leave `RISCV_RAM_BASE`/`RISCV_RAM_SIZE` as it is as reworking `libc` is outside the scope of this PR.